pegasus/src/homepage.rb

```diff
@@ -0,0 +1,130 @@
+        ]},
+
+        {type: "block", title: "homepage_slot_text_title_educators", text: "homepage_slot_text_blurb_educators", color1: "0, 148, 202", color2: "89, 185, 220", url: "/educate", image: "/shared/images/courses/logo_tall_teacher2.jpg", links: [
+          {text: "homepage_slot_text_link_elementary", url: "/educate/curriculum/elementary-school"},
+          {text: "homepage_slot_text_link_middle", url: "/educate/curriculum/middle-school"},
+          {text: "homepage_slot_text_link_high", url: "/educate/curriculum/high-school"}
+        ]},
+
+        {type: "block", title: "homepage_slot_text_title_hoc", text: "homepage_slot_text_blurb_hoc", color1: "0, 173, 188", color2: "89, 202, 211", url: "/learn", image: "/images/mc/2016_homepage_hocblock.jpg", links: [
+          {text: "homepage_slot_text_link_about_hoc", url: "https://hourofcode.com/"},
+          {text: "homepage_slot_text_link_host", url: "https://hourofcode.com/how-to"},
+          {text: "homepage_slot_text_link_hocserved", url: "/leaderboards"}
+        ]},
+
+        {type: "block", title: "homepage_slot_text_link_buy", text: "homepage_slot_text_blurb_advocates", color1: "185, 191, 21", color2: "209, 213, 103", url: "/help", image: "/shared/images/courses/logo_tall_map.jpg", links: [
+          {text: "homepage_slot_text_link_stats", url: "/promote"},
+          {text: "homepage_slot_text_link_administrators", url: "/administrators"},
+          {text: "homepage_slot_text_link_shop", url: "/shop"}
+        ]}
+      ]
+    else
+      [
+        {type: "blockshort", title: "homepage_slot_text_title_students", text: "homepage_slot_text_blurb_students", color1: "118, 101, 160", color2: "166, 155, 193", url: CDO.studio_url("/courses"), image: "/shared/images/courses/logo_tall_elementary.jpg"},
+        {type: "blockshort", title: "homepage_slot_text_title_educators", text: "homepage_slot_text_blurb_educators", color1: "0, 148, 202", color2: "89, 185, 220", url: CDO.studio_url("/courses?view=teacher"), image: "/shared/images/courses/logo_tall_teacher2.jpg"},
+        {type: "blockshort", title: "homepage_slot_text_title_hoc", text: "homepage_slot_text_blurb_hoc", color1: "0, 173, 188", color2: "89, 202, 211", url: "/learn", image: "/images/mc/2016_homepage_hocblock.jpg"},
+        {type: "blockshort", title: "studiobar_flappy_title", text: "studiobar_flappy_body", color1: "185, 191, 21", color2: "209, 213, 103", url: CDO.studio_url("/s/flappy/reset"), image: "/shared/images/courses/logo_tall_flappy.jpg"}
+      ]
+    end
+  end
+
+  def self.get_video
+    video = get_actions.find {|a| a[:type] == "video"}
+
+    {
+      video_code: video[:youtube_id],
+      download_path: video[:download_path],
+      facebook: {u: video[:facebook]},
+      twitter: {related: 'codeorg', text: video[:twitter]}
+    }
+  end
+
+  def self.get_census(request)
+    census_announcement = DCDO.get('census_announcement', nil)
+
```

Comment:
Not directly related, but I'm curious if we still need the census_announcement to be controlled with DCDO?  There isn't anything time sensitive about it anymore.

Reply:
That's a good question.  I think I'll leave it unchanged for this PR but it could be a good follow-on PR.